src/header/mod.rs

```diff
@@ -234,43 +232,45 @@ impl SphinxHeader {
     }
 
     fn blind_the_shared_secret(
-        shared_secret: SharedSecret,
+        shared_secret: PublicKey,
         blinding_factor: BlindingFactor,
```

Comment:
I wonder if we could just alias BlindingFactor to a normal private key (or remove it alltogether)

Reply:
I can turn it into a `StaticSecret` yes.

Reply:
Refactored in https://github.com/nymtech/sphinx/pull/94/commits/6899145f26fbab13afaa84e5d303e29ede919c80